pkg/snapstore/s3_snapstore.go

```diff
@@ -519,3 +536,25 @@ func isAWSConfigEmpty(config *awsCredentials) error {
 	}
 	return fmt.Errorf("aws s3 credentials: region, secretAccessKey or accessKeyID is missing")
 }
+
+// sseCredentials to hold fields for server-side encryption in I/O operations
+type sseCredentials struct {
+	SSECustomerAlgorithm string
+	SSECustomerKey       string
+	SSECustomerKeyMD5    string
+}
+
+// Checks for SSECustomerKey env var and creates the creds necessary
+func getSSECreds() (ssecreds sseCredentials) {
+	if SSECustomerKey, isSet := os.LookupEnv(sseCustomerKey); isSet {
+		SSECustomerKeyMD5Bytes := md5.Sum([]byte(SSECustomerKey))
+		SSECustomerKeyMD5 := base64.StdEncoding.EncodeToString(SSECustomerKeyMD5Bytes[:])
+		SSECustomerAlgorithm := sseCustomerAlgorithm
+		return sseCredentials{
+			SSECustomerAlgorithm: SSECustomerAlgorithm,
+			SSECustomerKey:       SSECustomerKey,
+			SSECustomerKeyMD5:    SSECustomerKeyMD5,
+		}
```

Comment:
`SSECustomerAlgorithm := sseCustomerAlgorithm` is unecessary? That could be directly be written as a part of the `return` statement.
```suggestion
		SSECustomerKeyMD5Bytes := md5.Sum([]byte(SSECustomerKey))
		SSECustomerKeyMD5 := base64.StdEncoding.EncodeToString(SSECustomerKeyMD5Bytes[:])
		return sseCredentials{
			SSECustomerAlgorithm: sseCustomerAlgorithm,
			SSECustomerKey:       SSECustomerKey,
			SSECustomerKeyMD5:    SSECustomerKeyMD5,
		}
```

Reply:
Yep. Agree. Fixed in the new commit

Reply:
Thanks.